op-node/p2p/peer_scorer.go

```diff
@@ -43,18 +44,50 @@ func NewScorer(peerGater PeerGater, peerStore Peerstore, metricer GossipMetricer
 	}
 }
 
+// ScoreBands holds the number of peers in each peer score band.
+// This is intended to be used with a [Metricer].
+type ScoreBands struct {
+	Graylist int
+	NoGossip int
+	NoPX     int
+	Friend   int
+}
+
+// Record records a peer score in the appropriate band.
+func (s *ScoreBands) Record(score float64) {
+	thresholds := NewPeerScoreThresholds()
+	if score <= thresholds.GraylistThreshold {
+		s.Graylist = s.Graylist + 1
+	} else if score <= thresholds.GossipThreshold {
+		s.NoGossip = s.NoGossip + 1
+	} else if score <= thresholds.AcceptPXThreshold {
+		s.NoPX = s.NoPX + 1
+	} else {
+		s.Friend = s.Friend + 1
+	}
+}
+
+// PlaceIntoBucket satisfies the [metrics.Bucketable] interface.
+func (s *ScoreBands) PlaceIntoBucket(gaugeVec *prometheus.GaugeVec) {
+	gaugeVec.WithLabelValues("graylist").Set(float64(s.Graylist))
+	gaugeVec.WithLabelValues("no_gossip").Set(float64(s.NoGossip))
+	gaugeVec.WithLabelValues("no_px").Set(float64(s.NoPX))
+	gaugeVec.WithLabelValues("friend").Set(float64(s.Friend))
+}
```

Comment:
That's fair, I was trying to interface things as much as possible to prevent having to shove `ScoreBands` into Metrics because that's an ugly solution. But, ok switching to that if this is too clunky.